### ISSUE TITLE
Managing Log Files

### ISSUE THREAD

**Issue by jshier** [logkit/logkit#2]
There doesn't seem to be any API for deleting, expiring, or limiting log files to a particular size. At the least I'd like to be able to manually delete the log files at runtime. However, this seems to break the file logging, likely because the `dispatch_io_t` created by the logger is broken when deleting the file, even if I recreate the log (empty) file on disk at the same location. Nor do I see any way to reinitialize the file logger. Any suggestions? 

Using Xcode 7 beta 3, with the Swift 2 branch, deploying back to iOS 7.

**Comment by justinpawela** [comment]
With the Swift 2 branch, I am aiming for LogKit 2.0 and would like to add more functionality to the included Endpoints. I had looked at rotating log files while building 1.0, but decided to push it off at the time as I didn't have an immediate need myself. I plan to come back to that.

In specific, I have been toying with an API like:

``` swift
LXRotatingFileEndpoint(fileURL: NSURL?, numberOfFiles: Int, maxFileSize: Int, ... )
```

I am happy to accept feedback on that API or another if other needs are not met by it.

I decided to wait because I couldn't decide if file size should just be checked at init time, or continuously monitored as an app runs (among other issues).

---

As a short-term solution, I believe a convenience initializer could be added to `LXLogFileEndpoint` along the lines of:

``` swift
public convenience init?(fileURL: NSURL?, resetFile: Bool, minimumLogLevel: LXLogLevel = .All, dateFormatter: NSDateFormatter = defaultDateFormatter, entryFormatter: LXLogEntryFormatter = defaultEntryFormatter) {
    if let URL = fileURL where resetFile {
        do {
            try NSFileManager.defaultManager().removeItemAtURL(URL)
        } catch {
            assertionFailure("File '\(fileURL?.path ?? String())' could not be removed")
        }
    }
    self.init(fileURL: fileURL, minimumLogLevel: minimumLogLevel, dateFormatter: dateFormatter, entryFormatter: entryFormatter)
}
```

That would at least allow the log file to be reset each time the app runs. I will try to work an API like this into the File Endpoint as I make improvements towards 2.0.

**Comment by justinpawela** [comment]
Closed by 4d86fea8ee3a6c60e27a1426684b1682c54d9ba0.

LogKit 2 contains enhanced file endpoints, including a new Rotating File Endpoint. The plain File Endpoint now has an option regarding appending vs starting fresh. And all endpoints in the File family have a method to reset the current log file at any time (although it's not exposed by the Logger, so you'll have to hold your own reference to the endpoint if you want to do that).

Thanks for you great feedback.
